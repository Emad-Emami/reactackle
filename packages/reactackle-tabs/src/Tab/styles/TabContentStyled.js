'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const base = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { paddingX, paddingY } = theme.reactackle.components.tabs.tab;

  return `
    padding: ${getValueString(paddingY)} ${getValueString(paddingX)};
  `;
};

export const TabContentStyled = styled.div`
  text-align: center;
  ${base};
`;

TabContentStyled.propTypes = propTypes;
TabContentStyled.displayName = 'TabContentStyled';
