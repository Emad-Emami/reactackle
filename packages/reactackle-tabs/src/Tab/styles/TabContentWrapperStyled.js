'use strict';

import styled from 'styled-components';

const content = () => `
  & > * {
    &,
    &:hover,
    &:focus {
      color: inherit;
      text-decoration: none;
      cursor: inherit;
    }
  }
`;

export const TabContentWrapperStyled = styled.div`
  flex-grow: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  ${content};
`;

TabContentWrapperStyled.displayName = 'TabContentWrapperStyled';
