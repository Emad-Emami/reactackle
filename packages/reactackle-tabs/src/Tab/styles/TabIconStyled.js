'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const size = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { width, height, imgSize } = theme.reactackle.components.tabs.icon;

  return `
    width: ${getValueString(width)};
    height: ${getValueString(height)};
    line-height: ${height}px;
    font-size: ${getValueString(imgSize)};
  `;
};

const style = ({ theme: themeFromProvider, colorScheme, selected }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const source = theme.reactackle.components.tabs.icon.style[colorScheme];

  return !selected
    ? `
      color: ${source.fontColor};
      
      &:hover {
        color: ${source.hover.fontColor};
      }
      
      &:focus {
        color: ${source.focus.fontColor};
      }`
    : `      
      &,
      &:hover,
      &:focus {
        color: ${source.selected.fontColor};
      }`;
};

export const TabIconStyled = styled.div`
  margin-left: auto;
  margin-right: auto;
  display: flex;
  ${size} ${style};
`;

TabIconStyled.propTypes = propTypes;
TabIconStyled.displayName = 'TabIconStyled';
