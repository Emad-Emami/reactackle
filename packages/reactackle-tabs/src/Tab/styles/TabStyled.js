'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  extractThemeOrDefault,
  getValueString,
  transition,
} from 'reactackle-core';

const propTypes = {
  colorScheme: PropTypes.oneOf(['light', 'dark']),
  /**
   * Define button theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  colorScheme: 'dark',
};

/** Prop Receivers */
const base = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const {
    fontSize,
    lineHeight,
    textTransform,
    minHeight,
    minWidth,
    maxWidth,
  } = theme.reactackle.components.tabs.tab;

  const { thickness } = theme.reactackle.components.tabs.bar;

  return `
    font-size: ${getValueString(fontSize)};
    line-height: ${lineHeight};
    text-transform: ${textTransform};
    min-height: ${getValueString(minHeight)};
    min-width: ${getValueString(minWidth)};
    max-width: ${getValueString(maxWidth)};
    padding-top: ${getValueString(thickness)};
    padding-bottom: ${getValueString(thickness)};
  `;
};

const style = ({ theme: themeFromProvider, colorScheme, selected }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const source = theme.reactackle.components.tabs.tab.style[colorScheme];

  return !selected
    ? `
      color: ${source.fontColor};
      background-color: ${source.backgroundColor};
      font-weight: ${source.fontWeight};
      opacity: ${source.opacity};
      cursor: pointer;

      &:hover {
        color: ${source.hover.fontColor};
        background-color: ${source.hover.backgroundColor};
        font-weight: ${source.hover.fontWeight};
        opacity: ${source.hover.opacity};
      }

      &:focus {
        color: ${source.focus.fontColor};
        background-color: ${source.focus.backgroundColor};
        font-weight: ${source.focus.fontWeight};
        opacity: ${source.focus.opacity};
      }`
    : `
      &,
      &:hover,
      &:focus {
        color: ${source.selected.fontColor};
        background-color: ${source.selected.backgroundColor};
        font-weight: ${source.selected.fontWeight};
        opacity: ${source.selected.opacity};
        cursor: default;
      }`;
};

const bar = ({ theme: themeFromProvider, selected, colorScheme }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { thickness } = theme.reactackle.components.tabs.bar,
    { backgroundColor } = theme.reactackle.components.tabs.bar.style[
      colorScheme
    ];

  return (
    selected &&
    `
      background-color: ${backgroundColor};
      height: ${getValueString(thickness)};
  `
  );
};

/** Styles */
export const TabStyled = styled.div`
  flex-grow: 1;
  display: flex;
  align-items: stretch;
  justify-content: center;
  position: relative;
  text-align: center;
  user-select: none;
  position: relative;
  ${base} ${style} ${transition('color, background-color, opacity')} &::after {
    display: block;
    content: '';
    width: 100%;
    position: absolute;
    bottom: 0;
    left: 0;
    z-index: 1;
    ${bar};
  }
`;

TabStyled.propTypes = propTypes;
TabStyled.defaultProps = defaultProps;
TabStyled.displayName = 'TabStyled';
