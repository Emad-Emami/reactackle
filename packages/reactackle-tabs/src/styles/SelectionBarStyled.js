'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  extractThemeOrDefault,
  getValueString,
  transition,
} from 'reactackle-core';

const propTypes = {
  colorScheme: PropTypes.oneOf(['default', 'light', 'dark']),
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  colorScheme: 'dark',
};

const style = ({ theme: themeFromProvider, colorScheme }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const path = theme.reactackle.components.tabs.bar,
    { backgroundColor } = path.style[colorScheme],
    { thickness } = path;

  return `
    background-color: ${backgroundColor};
    height: ${getValueString(thickness)};
  `;
};

export const SelectionBarStyled = styled.div`
  display: block;
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 2;
  width: 100%;
  ${style};
  ${transition('transform')};
`;

SelectionBarStyled.propTypes = propTypes;
SelectionBarStyled.defaultProps = defaultProps;
SelectionBarStyled.displayName = 'SelectionBarStyled';
