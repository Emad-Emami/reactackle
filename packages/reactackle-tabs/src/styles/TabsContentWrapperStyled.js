'use strict';

import styled from 'styled-components';

export const TabsContentWrapperStyled = styled.div`
  display: flex;
  align-items: stretch;
  position: relative;
  flex-grow: 1;
  overflow: hidden;
`;

TabsContentWrapperStyled.displayName = 'TabsContentWrapperStyled';
