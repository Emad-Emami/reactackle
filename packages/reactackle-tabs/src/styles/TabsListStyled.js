'use strict';

import styled from 'styled-components';
import { transition } from 'reactackle-core';

export const TabsListStyled = styled.div`
  display: flex;
  flex-grow: 1;
  justify-content: center;
  ${transition('transform')};
`;

TabsListStyled.displayName = 'TabsListStyled';
