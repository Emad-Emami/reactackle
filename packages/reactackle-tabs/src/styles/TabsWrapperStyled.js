'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';
import { extractThemeOrDefault } from 'reactackle-core';

const propTypes = {
  colorScheme: PropTypes.oneOf(['light', 'dark']),
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  colorScheme: 'dark',
};

const style = ({ theme: themeFromProvider, colorScheme }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { backgroundColor } = theme.reactackle.components.tabs.tabs.style[
      colorScheme
    ],
    { fontColor } = theme.reactackle.components.tabs.tab.style[colorScheme];

  const fontFamily =
    theme.reactackle.fontFamily[theme.reactackle.body.fontFamily];

  return `
    background-color: ${backgroundColor};
    color: ${fontColor};
    
    &,
    & *,
    *::after,
    *::before {
      font-family: ${fontFamily};
      box-sizing: border-box;
    }
  `;
};

export const TabsWrapperStyled = styled.div`
  display: flex;
  align-items: stretch;
  position: relative;
  ${style};
`;

TabsWrapperStyled.propTypes = propTypes;
TabsWrapperStyled.defaultProps = defaultProps;
TabsWrapperStyled.displayName = 'TabsWrapperStyled';
