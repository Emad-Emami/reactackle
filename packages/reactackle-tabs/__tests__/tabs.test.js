'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { setProps, findAllByType, returnNull } from 'reactackle-core';
import { Tabs } from '../src';
import { TabStyled } from '../src/Tab/styles/TabStyled';

jest.mock('react-dom');

describe('<Tabs/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(<Tabs />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop selected set 1', () => {
    const tree = renderer.create(
      <Tabs
        selected={1}
        tabs={[
        { text: 'Test' },
        { text: 'Test' },
        { text: 'Test' },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop linkComponent passed', () => {
    const tree = renderer.create(
      <Tabs
        linkComponent={returnNull}
        tabs={[
        { text: 'Test', linkHref: '/' },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop icon', () => {
    const tree = renderer.create(
      <Tabs
        tabs={[
          { text: 'Test', icon: { name: 'anchor' } },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop text is empty', () => {
    const tree = renderer.create(
      <Tabs
        tabs={[
          { text: '' },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if prop ', () => {
    const tree = renderer.create(
      <Tabs
        tabs={[
          { text: 'Test1', linkHref: 'test1' },
          { text: 'Test2', linkHref: 'test2' },
        ]}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if change tab', () => {
    const mockFn = jest.fn(),
      component = renderer.create(
        <Tabs
          onChange={mockFn}
          tabs={[
            { text: 'Test 1', icon: { name: 'anchor' } },
            { text: 'Test 2', icon: { name: 'anchor' } },
          ]}
        />,
      );

    expect(component.toJSON()).toMatchSnapshot();

    const tabs = findAllByType(component, TabStyled);
    tabs[1].props.onClick();

    expect(mockFn).toHaveBeenCalledWith({ value: 1 });
    expect(component.toJSON()).toMatchSnapshot();
  });


  it('renders correctly if prop selected set after render', () => {
    const tree = renderer.create(
      <Tabs
        selected={2}
        tabs={[
        { text: 'Test' },
        { text: 'Test' },
        { text: 'Test' },
        ]}
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
    setProps(tree, { selected: 1 });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
