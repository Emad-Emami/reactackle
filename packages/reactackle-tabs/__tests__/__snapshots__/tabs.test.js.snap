// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs/> renders correctly if change tab 1`] = `
<Tabs
  colorScheme="dark"
  linkComponent={[Function]}
  onChange={[Function]}
  selected={0}
  tabs={
    Array [
      Object {
        "icon": <svg
          viewBox="0 0 24 24"
    >
          <path
                d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
          />
    </svg>,
        "text": "Test 1",
      },
      Object {
        "icon": <svg
          viewBox="0 0 24 24"
    >
          <path
                d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
          />
    </svg>,
        "text": "Test 2",
      },
    ]
  }
>
  <TabsWrapperStyled
    colorScheme="dark"
  >
    <div
      className="TabsWrapperStyled-dqdZaj irKElO"
    >
      <TabsContentWrapperStyled>
        <div
          className="TabsContentWrapperStyled-gZFwCM bqxSvP"
        >
          <TabsListStyled>
            <div
              className="TabsListStyled-hCaYrA dBLEcs"
            >
              <Tab
                colorScheme="dark"
                disabled={false}
                icon={
                  <svg
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
                    />
                  </svg>
                }
                index={0}
                isSelected={true}
                key="0"
                linkComponent={[Function]}
                linkHref=""
                onSelect={[Function]}
                text="Test 1"
              >
                <TabStyled
                  colorScheme="dark"
                  onClick={[Function]}
                  selected={true}
                >
                  <div
                    className="TabStyled-kyRlfB vASeN"
                    onClick={[Function]}
                    selected={true}
                  >
                    <TabContentWrapperStyled>
                      <div
                        className="TabContentWrapperStyled-gkOJpN bJEhdI"
                      >
                        <TabContentStyled>
                          <div
                            className="TabContentStyled-ksDzLH ieZSTT"
                          >
                            <TabIconStyled
                              colorScheme="dark"
                              selected={true}
                            >
                              <div
                                className="TabIconStyled-clnPsW jfCxxK"
                                selected={true}
                              >
                                <svg
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
                                  />
                                </svg>
                              </div>
                            </TabIconStyled>
                            <TabTextStyled
                              hasIcon={true}
                            >
                              <div
                                className="TabTextStyled-hKBktu hyUoMC"
                              >
                                Test 1
                              </div>
                            </TabTextStyled>
                          </div>
                        </TabContentStyled>
                      </div>
                    </TabContentWrapperStyled>
                  </div>
                </TabStyled>
              </Tab>
              <Tab
                colorScheme="dark"
                disabled={false}
                icon={
                  <svg
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
                    />
                  </svg>
                }
                index={1}
                isSelected={false}
                key="1"
                linkComponent={[Function]}
                linkHref=""
                onSelect={[Function]}
                text="Test 2"
              >
                <TabStyled
                  colorScheme="dark"
                  onClick={[Function]}
                  selected={false}
                >
                  <div
                    className="TabStyled-kyRlfB bgEHWe"
                    onClick={[Function]}
                    selected={false}
                  >
                    <TabContentWrapperStyled>
                      <div
                        className="TabContentWrapperStyled-gkOJpN bJEhdI"
                      >
                        <TabContentStyled>
                          <div
                            className="TabContentStyled-ksDzLH ieZSTT"
                          >
                            <TabIconStyled
                              colorScheme="dark"
                              selected={false}
                            >
                              <div
                                className="TabIconStyled-clnPsW faBOEm"
                                selected={false}
                              >
                                <svg
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
                                  />
                                </svg>
                              </div>
                            </TabIconStyled>
                            <TabTextStyled
                              hasIcon={true}
                            >
                              <div
                                className="TabTextStyled-hKBktu hyUoMC"
                              >
                                Test 2
                              </div>
                            </TabTextStyled>
                          </div>
                        </TabContentStyled>
                      </div>
                    </TabContentWrapperStyled>
                  </div>
                </TabStyled>
              </Tab>
            </div>
          </TabsListStyled>
        </div>
      </TabsContentWrapperStyled>
    </div>
  </TabsWrapperStyled>
</Tabs>
`;

exports[`<Tabs/> renders correctly if change tab 2`] = `
<Tabs
  colorScheme="dark"
  linkComponent={[Function]}
  onChange={[Function]}
  selected={0}
  tabs={
    Array [
      Object {
        "icon": <svg
          viewBox="0 0 24 24"
    >
          <path
                d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
          />
    </svg>,
        "text": "Test 1",
      },
      Object {
        "icon": <svg
          viewBox="0 0 24 24"
    >
          <path
                d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
          />
    </svg>,
        "text": "Test 2",
      },
    ]
  }
>
  <TabsWrapperStyled
    colorScheme="dark"
  >
    <div
      className="TabsWrapperStyled-dqdZaj irKElO"
    >
      <TabsContentWrapperStyled>
        <div
          className="TabsContentWrapperStyled-gZFwCM bqxSvP"
        >
          <TabsListStyled>
            <div
              className="TabsListStyled-hCaYrA dBLEcs"
            >
              <Tab
                colorScheme="dark"
                disabled={false}
                icon={
                  <svg
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
                    />
                  </svg>
                }
                index={0}
                isSelected={false}
                key="0"
                linkComponent={[Function]}
                linkHref=""
                onSelect={[Function]}
                text="Test 1"
              >
                <TabStyled
                  colorScheme="dark"
                  onClick={[Function]}
                  selected={false}
                >
                  <div
                    className="TabStyled-kyRlfB bgEHWe"
                    onClick={[Function]}
                    selected={false}
                  >
                    <TabContentWrapperStyled>
                      <div
                        className="TabContentWrapperStyled-gkOJpN bJEhdI"
                      >
                        <TabContentStyled>
                          <div
                            className="TabContentStyled-ksDzLH ieZSTT"
                          >
                            <TabIconStyled
                              colorScheme="dark"
                              selected={false}
                            >
                              <div
                                className="TabIconStyled-clnPsW faBOEm"
                                selected={false}
                              >
                                <svg
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
                                  />
                                </svg>
                              </div>
                            </TabIconStyled>
                            <TabTextStyled
                              hasIcon={true}
                            >
                              <div
                                className="TabTextStyled-hKBktu hyUoMC"
                              >
                                Test 1
                              </div>
                            </TabTextStyled>
                          </div>
                        </TabContentStyled>
                      </div>
                    </TabContentWrapperStyled>
                  </div>
                </TabStyled>
              </Tab>
              <Tab
                colorScheme="dark"
                disabled={false}
                icon={
                  <svg
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
                    />
                  </svg>
                }
                index={1}
                isSelected={true}
                key="1"
                linkComponent={[Function]}
                linkHref=""
                onSelect={[Function]}
                text="Test 2"
              >
                <TabStyled
                  colorScheme="dark"
                  onClick={[Function]}
                  selected={true}
                >
                  <div
                    className="TabStyled-kyRlfB vASeN"
                    onClick={[Function]}
                    selected={true}
                  >
                    <TabContentWrapperStyled>
                      <div
                        className="TabContentWrapperStyled-gkOJpN bJEhdI"
                      >
                        <TabContentStyled>
                          <div
                            className="TabContentStyled-ksDzLH ieZSTT"
                          >
                            <TabIconStyled
                              colorScheme="dark"
                              selected={true}
                            >
                              <div
                                className="TabIconStyled-clnPsW jfCxxK"
                                selected={true}
                              >
                                <svg
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
                                  />
                                </svg>
                              </div>
                            </TabIconStyled>
                            <TabTextStyled
                              hasIcon={true}
                            >
                              <div
                                className="TabTextStyled-hKBktu hyUoMC"
                              >
                                Test 2
                              </div>
                            </TabTextStyled>
                          </div>
                        </TabContentStyled>
                      </div>
                    </TabContentWrapperStyled>
                  </div>
                </TabStyled>
              </Tab>
            </div>
          </TabsListStyled>
        </div>
      </TabsContentWrapperStyled>
    </div>
  </TabsWrapperStyled>
</Tabs>
`;

exports[`<Tabs/> renders correctly if prop  1`] = `
<div
  className="TabsWrapperStyled-dqdZaj irKElO"
>
  <div
    className="TabsContentWrapperStyled-gZFwCM bqxSvP"
  >
    <div
      className="TabsListStyled-hCaYrA dBLEcs"
    >
      <a
        href="test1"
        onClick={[Function]}
      >
        <div
          className="TabStyled-kyRlfB vASeN"
          onClick={[Function]}
          selected={true}
        >
          <div
            className="TabContentWrapperStyled-gkOJpN bJEhdI"
          >
            <div
              className="TabContentStyled-ksDzLH ieZSTT"
            >
              <div
                className="TabTextStyled-hKBktu dIRLXI"
              >
                Test1
              </div>
            </div>
          </div>
        </div>
      </a>
      <a
        href="test2"
        onClick={[Function]}
      >
        <div
          className="TabStyled-kyRlfB bgEHWe"
          onClick={[Function]}
          selected={false}
        >
          <div
            className="TabContentWrapperStyled-gkOJpN bJEhdI"
          >
            <div
              className="TabContentStyled-ksDzLH ieZSTT"
            >
              <div
                className="TabTextStyled-hKBktu dIRLXI"
              >
                Test2
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop icon 1`] = `
<div
  className="TabsWrapperStyled-dqdZaj irKElO"
>
  <div
    className="TabsContentWrapperStyled-gZFwCM bqxSvP"
  >
    <div
      className="TabsListStyled-hCaYrA dBLEcs"
    >
      <div
        className="TabStyled-kyRlfB vASeN"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabIconStyled-clnPsW jfCxxK"
              selected={true}
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M20,5.6C20,4.7,19.3,4,18.4,4H5.6C4.7,4,4,4.7,4,5.6v9.6c0,0.9,0.7,1.6,1.6,1.6h11.2L20,20L20,5.6z"
                />
              </svg>
            </div>
            <div
              className="TabTextStyled-hKBktu hyUoMC"
            >
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop linkComponent passed 1`] = `
<div
  className="TabsWrapperStyled-dqdZaj irKElO"
>
  <div
    className="TabsContentWrapperStyled-gZFwCM bqxSvP"
  >
    <div
      className="TabsListStyled-hCaYrA dBLEcs"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="TabStyled-kyRlfB vASeN"
          onClick={[Function]}
          selected={true}
        >
          <div
            className="TabContentWrapperStyled-gkOJpN bJEhdI"
          >
            <div
              className="TabContentStyled-ksDzLH ieZSTT"
            >
              <div
                className="TabTextStyled-hKBktu dIRLXI"
              >
                Test
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop selected set 1 1`] = `
<div
  className="TabsWrapperStyled-dqdZaj irKElO"
>
  <div
    className="TabsContentWrapperStyled-gZFwCM bqxSvP"
  >
    <div
      className="TabsListStyled-hCaYrA dBLEcs"
    >
      <div
        className="TabStyled-kyRlfB bgEHWe"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabTextStyled-hKBktu dIRLXI"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled-kyRlfB vASeN"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabTextStyled-hKBktu dIRLXI"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled-kyRlfB bgEHWe"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabTextStyled-hKBktu dIRLXI"
            >
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop selected set after render 1`] = `
<div
  className="TabsWrapperStyled-dqdZaj irKElO"
>
  <div
    className="TabsContentWrapperStyled-gZFwCM bqxSvP"
  >
    <div
      className="TabsListStyled-hCaYrA dBLEcs"
    >
      <div
        className="TabStyled-kyRlfB bgEHWe"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabTextStyled-hKBktu dIRLXI"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled-kyRlfB bgEHWe"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabTextStyled-hKBktu dIRLXI"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled-kyRlfB vASeN"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabTextStyled-hKBktu dIRLXI"
            >
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop selected set after render 2`] = `
<div
  className="TabsWrapperStyled-dqdZaj irKElO"
>
  <div
    className="TabsContentWrapperStyled-gZFwCM bqxSvP"
  >
    <div
      className="TabsListStyled-hCaYrA dBLEcs"
    >
      <div
        className="TabStyled-kyRlfB bgEHWe"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabTextStyled-hKBktu dIRLXI"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled-kyRlfB vASeN"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabTextStyled-hKBktu dIRLXI"
            >
              Test
            </div>
          </div>
        </div>
      </div>
      <div
        className="TabStyled-kyRlfB bgEHWe"
        onClick={[Function]}
        selected={false}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          >
            <div
              className="TabTextStyled-hKBktu dIRLXI"
            >
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly if prop text is empty 1`] = `
<div
  className="TabsWrapperStyled-dqdZaj irKElO"
>
  <div
    className="TabsContentWrapperStyled-gZFwCM bqxSvP"
  >
    <div
      className="TabsListStyled-hCaYrA dBLEcs"
    >
      <div
        className="TabStyled-kyRlfB vASeN"
        onClick={[Function]}
        selected={true}
      >
        <div
          className="TabContentWrapperStyled-gkOJpN bJEhdI"
        >
          <div
            className="TabContentStyled-ksDzLH ieZSTT"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tabs/> renders correctly with default props 1`] = `
<div
  className="TabsWrapperStyled-dqdZaj irKElO"
>
  <div
    className="TabsContentWrapperStyled-gZFwCM bqxSvP"
  >
    <div
      className="TabsListStyled-hCaYrA dBLEcs"
    />
  </div>
</div>
`;
